frontend/src/components/management/WorkflowAssignment.tsx

```diff
@@ -0,0 +1,448 @@
+                    style={{ padding: '1em' }}
+                >
+                    <Button disabled={action.alreadyQueued.length > 0} onClick={async () => {
+                        if (!selectedProject) return
+                        const project = projectByID[selectedProject]
+                        try {
+                            await dispatch(api.samples.addSamplesToStudy(exceptedSampleIDs, defaultSelection, selectedProject, action.study.letter, action.stepOrder, serializeFilterParamsWithDescriptions(filters)))
+                            dispatch(notifySuccess({
+                                id: `LabworkSamples_${action.study.id}_${action.stepOrder}`,
+                                title: "Samples queued to workflow",
+                                description: `Successfully queued samples to study ${action.study.letter} at step "${action.step} for project ${project.name}"`
+                            }))
+                            await refreshActions()
+                        } catch (error) {
+                            const errors: string[] | undefined = error.data?.['queue_sample_to_study_workflow']
+                            dispatch(notifyError({
+                                id: `LabworkSamples_${action.study.id}_${action.stepOrder}`,
+                                title: "Error queuing samples to workflow",
+                                description: errors ? `${errors[0]}${errors[0].endsWith('.') ? '' : '.'}${errors.length > 1 ? ' And ' + (errors.length - 1) + ' more errors...' : ''}` : `Could not queue samples to study ${action.study.letter} at step "${action.step}" for project ${project.name}".`,
+                                duration: 5
+                            }))
+                        }
+                    }} type="primary">{`${action.stepOrder}. ${action.step}`}</Button>
+                </Popover>
+            })
+            result.push(
+                <Collapse key={'actions'} accordion items={[
+                    {
+                        key: 'dequeue',
+                        label: `Dequeue Options (${projectDequeueActions.length})`,
+                        children: <Flex wrap gap={"small"}>{projectDequeueActions}</Flex>,
+                        classNames: { body: 'labwork-samples-actions' },
+                    },
+                    {
+                        key: 'queue',
+                        label: `Queue Options (${projectQueueActions.length})`,
+                        children: <Flex wrap gap={"small"}>{projectQueueActions}</Flex>,
+                        classNames: { body: 'labwork-samples-actions' },
+                    }
+                ]}/>
+            )
+        }
+        return result
+    }, [commonProjects, defaultSelection, dequeueActions, dispatch, exceptedSampleIDs, filters, isFetching, projectByID, queueActions, refreshActions, sampleByID, selectedProject, selectedStudyWorkflow, studyWorkflowsByProject])
```

Comment:
not sure how valuable is the useMemo in this case ... each time basically anything change in the page you have to calculate it again ... Might be more useful to have more targeted ones. To me it looks like an umbrella against unintended re-renders. Might be more efficient to fix the unintended re-renders at the source (unnecessary useAppSelectors ?).

Reply:
It's more of a convenience of being in a function than useMemo ;)
It let's me take advantage of return to exit out with a component result sooner.

Reply:
eheh you might be the only one i know thinking a useMemo with 10 dependencies is easier to understand than an if ;). that fine you can keep it ;).